docs/tutorials/index-to-local-csv-files.md

```diff
@@ -0,0 +1,217 @@
+Similarly, the `syncIntervalBlocks` configuration defines how many blocks have to be ingested, before creating a new CSV file, when the indexing process has reached the blockchain's head and is in sync with it.
+:::
+
+### Data indexing
+
+All the indexing logic is defined in the file named processor.ts, so let's open it, and edit the `EvmBatchProcessor` class configuration, to request data for the right smart contract, and the right EVM log:
+
+```typescript
+export const contractAddress =
+  "0x7D1AfA7B718fb893dB30A3aBc0Cfc608AaCfeBB0".toLowerCase();
+
+const processor = new EvmBatchProcessor()
+  .setDataSource({
+    archive: lookupArchive("eth-mainnet"),
+    chain: process.env.RPC_ENDPOINT,
+  })
+  .addLog(contractAddress, {
+    filter: [[events.Transfer.topic]],
+    data: {
+      evmLog: {
+        topics: true,
+        data: true,
+      },
+    } as const,
+  });
+```
+
+:::warning
+Note: the RPC_ENDPOINT environment variable is used, so make sure to edit the `.env` file and use a valid url of an Ethereum RPC node, e.g.:
+
+```bash
+DB_NAME=squid
+DB_PORT=23798
+GQL_PORT=4350
+# JSON-RPC node endpoint, both wss and https endpoints are accepted
+RPC_ENDPOINT="https://rpc.ankr.com/eth"
+```
+:::
+
+Let's then define the logic to process a batch of EVM log data, and save it in the CSV files.
+
+A double loop is necessary to explore all the blocks in each batch, and the items in each block. In the innermost loop, it's necessary to check that the items are actually EVM logs (this favors TypeScript typings, as [different kind of items have access to different fields](/evm-indexing/index.md#overview-and-the-data-model)), that they have been generated by the right address and that they have the right topic signature.
+
+It's then possible to decode the event and prepare an object with the right data structure, which is added to an array.
+
+Outside both loops, I perform a single operation to save all the objects in the previously prepared array. This batch operation helps by increasing the overall performance.
+Here is a short summary of the logic:
+
+```typescript
+processor.run(db, async (ctx) => {
+  const transfers: Record[] = [];
+  let decimals = 18;
+  for (let block of ctx.blocks) {
+    for (let item of block.items) {
+      if (item.address !== contractAddress) continue;
+      if (item.kind !== "evmLog") continue;
+      if (item.evmLog.topics[0] !== events.Transfer.topic) continue;
+
+      const { from, to, value } = events.Transfer.decode(item.evmLog);
+      transfers.push({
+        blockNumber: block.header.height,
+        timestamp: new Date(block.header.timestamp),
+        contractAddress: item.address,
+        from: from.toLowerCase(),
+        to: to.toLowerCase(),
+        amount: BigDecimal(value.toBigInt(), decimals).toNumber(),
+      });
+    }
+  }
+
+  ctx.store.Transfers.writeMany(transfers);
```

Comment:
no need to use writeMany, it's the same as
for () {
  ctx.store.Transfers.write(transfer)
}

Reply:
Thanks for the tip, but it's still one line vs a for loop, I prefer it 🤷

Reply:
you already have for loop

Reply:
replace .push with .write